src/wazuh_db/wdb_agent.c

```diff
@@ -375,181 +287,201 @@ int wdb_remove_agent_db(int id, const char * name) {
 
 /* Get an array containing the ID of every agent (except 0), ended with -1 */
 int* wdb_get_all_agents() {
-    int i;
-    int n = 1;
-    int *array;
-    sqlite3_stmt *stmt = NULL;
-
-    if (!(array = malloc(sizeof(int)))) {
-        merror("wdb_get_all_agents(): memory error");
-        return NULL;
-    }
-
-    if (wdb_open_global() < 0) {
-        free(array);
-        return NULL;
-    }
-
-    if (wdb_prepare(wdb_global, SQL_SELECT_AGENTS, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        wdb_close_global();
-        free(array);
+    int i = 0;
+    int n = 0;
+    int *array = NULL;
+    char *json_string = NULL;
+    cJSON *elem = NULL;
+    cJSON *name = NULL;
+    cJSON *root = NULL;
+    char wdbquery[OS_BUFFER_SIZE] = "";
+    char wdboutput[OS_BUFFER_SIZE] = "";
+    int wdb_sock = -1;
+
+    snprintf(wdbquery, OS_BUFFER_SIZE,"%s", global_db_queries[SQL_SELECT_AGENTS]);
+    wdbc_query_ex(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
```

Comment:
Shouldn't we check here if this returns an error?

Reply:
Now it checks the result